docs/decisions/0002-lti-1p3-variables.rst

```diff
@@ -0,0 +1,123 @@
+LTI 1.3 variable definition
+---------------------------
+
+Status
+======
+
+Provisional
+
+Context
+=======
+
+The LTI 1.3 specification defines multiple variables related to course, context, problem, student id and PII.
+
+The goal of this ADR is to provide well defined mappings from variables available in the platform and
+variables required by LTI 1.3 launches.
+
+Decisions
+=========
+
+The following section go over required and optional variables in each supported claim in the LTI 1.3 implementation.
+This ADR should be updated when new claims are added.
+
+Deployment ID
+~~~~~~~~~~~~~
++---------------------------------------------------------+------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| LTI 1.3 variable name                                   | Open edX         | Reasoning                                                                                                                                                                                                                       |
++---------------------------------------------------------+------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| https://purl.imsglobal.org/spec/lti/claim/deployment_id | "1" (fixed value)| We're using single-tenant deployments, were each key is used for a single XBlock. Even when moving up to the course level, we'll can keep this setup since multi-tenancy will only be required for instance wide configuration. |
++---------------------------------------------------------+------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Comment:
I'm not sure I understand this. Do you mean that instead of one global client_id with multiple different deployments within that, we have multiple client_ids each with a single deployment?

Reply:
On LTI, Client ID > Deployment > Resource link (those are 3 separate things).

The client ID is coupled with a single key pair in the platform, and each Client ID can have multiple deployments inside the LMS. I've opted out of that approach for the LTI block implementation because it introduces a level of complexity not needed for the XBlock level implementation.

See http://www.imsglobal.org/spec/lti/v1p3/#single-tenant-tool-registered-and-deployed-once for more details on how LTI defines what a deployment is.

Reply:
Sure, so what I understand is, we aren't using the concept of deployments at all because there isn't really any use for it yet in the platform.

Reply:
Yes, we're not using it, but we can easily support it. 
`Deployment` in LTI is basically an abstraction that allows us to use the same client id/key pair to deploy isolated tools (for different courses, for example).

Reply:
I am certain I don't understand this yet, because the LTI spec is kind of impenetrable.  But [§3.1.3 Tool Deployment](http://www.imsglobal.org/spec/lti/v1p3/#tool-deployment) says:
> A platform MUST generate a unique deployment_id for each tool it integrates with.
Doesn't that mean that `"1"` isn't enough?

Reply:
Hmm, never mind, the diagram a few paragraphs below explicitly shows three tools all using the same deployment id.